pandas/_libs/src/vendored/ujson/python/objToJSON.c

```diff
@@ -1293,7 +1301,8 @@ char **NpyArr_encodeLabels(PyArrayObject *labels, PyObjectEncoder *enc,
         }
 
         int is_datetimelike = 0;
-        npy_int64 nanosecVal;
+        npy_int64 i8date;
+        NPY_DATETIMEUNIT dateUnit = NPY_FR_ns;
```

Comment:
Is the idea here that this should default to NS for things that are stored within object arrays?

Reply:
It's for the other stuff like timedeltas and dates. We use nanosecond reso there.

I didn't try to fix those cases as well.

Reply:
What do you think it would take to handle those consistently? I don't know how common this would happen but I guess its strange to only do this for numpy-typed arrays and not object arrays that may contain datetimes

Reply:
knee-jerk i think that for object-dtype we should just call `str(x)` on everything and call it a day

Reply:
The object stuff definitely adds a lot of complexity that may not be worth it, but that would be a breaking change to just pass as a str. Particularly for nested containers within an object array that would not work

Reply:
I think it might work for object arrays, already.

I haven't checked yet, I'm not on the right branch. Will update later in the day.

Reply:
Well, it turns out that this didn't work already. Sorry for the confusion.

I've fixed this in the latest commit.